addons/mail/models/mail_followers.py

```diff
@@ -145,3 +70,166 @@ def unlink(self):
         ('mail_followers_res_channel_res_model_id_uniq', 'unique(res_model,res_id,channel_id)', 'Error, a channel cannot follow twice the same object.'),
         ('partner_xor_channel', 'CHECK((partner_id IS NULL) != (channel_id IS NULL))', 'Error: A follower must be either a partner or a channel (but not both).')
     ]
+
+    # --------------------------------------------------
+    # Private tools methods to fetch followers data
+    # --------------------------------------------------
+
+    def _get_followers_data(self, res_model, res_ids, pids, cids):
+        where_params = (res_model, res_ids, pids, cids)
+        query = """
+SELECT fol.id, fol.res_id, fol.partner_id, fol.channel_id, array_agg(subtype.id)
+FROM mail_followers fol
+LEFT JOIN mail_followers_mail_message_subtype_rel fol_rel ON fol_rel.mail_followers_id = fol.id
+LEFT JOIN mail_message_subtype subtype ON subtype.id = fol_rel.mail_message_subtype_id
+WHERE fol.res_model = %s AND fol.res_id = ANY(%s) AND
+    (fol.partner_id = ANY(%s) OR fol.channel_id = ANY(%s))
+GROUP BY fol.id"""
```

Comment:
The query uses a lot of `... = ANY(%s)`.  Why don't you use `... IN %s` instead?
Just be careful to serialize ids as tuples instead of lists. This enables the method to accept any collection of ids.

Reply:
According to https://stackoverflow.com/questions/31191507/how-to-use-any-instead-of-in-in-a-where-clause-with-rails/31192557#31192557 actually using ANY is a newer version of IN. Do you think using IN is necessary ?

Reply:
About why I suggest `IN` instead of `ANY`:
- In the ORM we never use `ANY`, we systematically use `IN`. That's why `ANY` looks weird to me. But that's just me.
- To me, `left IN %s` is clearer than `left = ANY(%s)`, but that's personal taste.
- In my branch on subqueries, I used explicitly the fact that `IN` can take a tuple or a subquery on its right.
- Special cases with empty collections: `ANY(ARRAY [])` crashes just like `IN ()`.

In case you decide to keep `ANY`, then imagine that the caller may want to use any collection (list, set, dict) as input. You can provide much flexibility by wrapping parameters in `list(...)`.

Reply:
Ok, let's go for IN then :) .